easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -87,6 +87,48 @@ def setup_cmake_env(tc):
     setvar("CMAKE_LIBRARY_PATH", library_paths)
 
 
+def setup_cmake_env_python_hints(cmake_version):
+    """Convenience function to set CMake hints for FindPython[_2/3] as environment variables.
+    Needed to avoid wrong Python being picked up by CMake when not called directly by EasyBuild but as step in a
+    build and no option is provided to set custom CMake variables.
+    """
+    cmake_version = det_cmake_version()
+    if LooseVersion(cmake_version) < '3.12':
+        raise EasyBuildError("Setting Python hints for CMake requires CMake version 3.12 or newer")
+    python_root = get_software_root('Python')
+    if python_root:
+        python_version = LooseVersion(get_software_version('Python'))
+        setvar('Python_ROOT_DIR', python_root)
+        if python_version >= "3":
+            setvar('Python3_ROOT_DIR', python_root)
+        else:
+            setvar('Python2_ROOT_DIR', python_root)
+
+
+def get_cmake_python_config_dict():
+    """Get a dictionary with the CMake configuration options for Python hints."""
+    options = {}
+    python_root = get_software_root('Python')
+    if python_root:
+        python_version = LooseVersion(get_software_version('Python'))
+        python_exe = os.path.join(python_root, 'bin', 'python')
+        # This is required for (deprecated) `find_package(PythonInterp ...)`
+        options['PYTHON_EXECUTABLE'] = python_exe
+        # Ensure that both `find_package(Python) and find_package(Python2/3)` work as intended
+        options['Python_EXECUTABLE'] = python_exe
+        if python_version >= "3":
+            options['Python3_EXECUTABLE'] = python_exe
+        else:
+            options['Python2_EXECUTABLE'] = python_exe
+    return options
+
+
+def get_cmake_python_config_str():
+    """Get a string with the CMake configuration options for Python hints."""
+    options = get_cmake_python_config_dict()
+    return ' '.join(['-D%s=%s' % (key, value) for key, value in options.items()])
```

Comment:
No need to create a list first:
```suggestion
    return ' '.join('-D%s=%s' % (key, value) for key, value in options.items())
```